pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/FilterOperatorUtils.java

```diff
@@ -25,6 +25,10 @@
 
 
 public class FilterOperatorUtils {
+
+  // empirically derived max cardinality value; can be fine tuned/learnt in future iterations
+  private static final int DEFAULT_MAX_CARDINALITY = 1000_000;
```

Comment:
Should we make this configurable (if we expect to tune this in future, or reset it to a different value depending on use case)